### PR TITLE
feat(api): invert management of hw control thread

### LINKED ISSUE #4870: Invert thread control for hardware controller and synchronous adapters
For better correctness and less buggy execution, we should make hardware controllers live in their own threads.

Currently, every time you make a synchronous adapter it spawns a new thread and a new asyncio loop to run its hardware controller in.

Instead, we should make a thread something a hardware controller has, and a synchronous adapter does not own its own thread. This will also fix the issues we have with synchronous adapter not properly inheriting from threading.Thread like #4609

### PR DESCRIPTION
**Pull request by b-cooper**
## overview

This branch removes thread management from `SynchronousAdapter`'s responsibilities and instead opts to manage a singular thread for all hw control concerns in a new class call `ThreadManager`.  

Closes #4870

## changelog

`ThreadManager` currently only wraps instances of `opentrons.hardware_control.API` and provides an threadsafe interface to call into all of its members and children.

Many instances of `ThreadManager` are wrapped by instances of `SynchronousAdapter` which further abstracts the interface to hardware control members by masking any call to a hardware-controller-owned coroutine or coroutine function as a synchronous function in the calling thread. `SynchronousAdapter` looks for a `_loop` member on the object that it wraps, and will pass all coroutines to that loop to execute. 

## review requests

This code touches nearly all instantiation methods of the hardware control classes. Because of this, it is necessary to test all entry points and ensure they behave as expected.

### CODE REVIEW

File: api/src/opentrons/legacy_api/instruments/pipette.py [hunk 1]
```diff
@@ -1127,7 +1127,6 @@ def drop_tip(self, location=None, home_after=True):
             if 'rack' in location.get_parent().get_type():
                 half_tip_length = self._tip_length * self._return_tip_height
                 location = location.top(-half_tip_length)
-                print(f'v1 location: {location}')
```
Comment (b-cooper) [inline api/src/opentrons/legacy_api/instruments/pipette.py]: Just noticed this while grepping, figured I'd clean it out.

File: api/src/opentrons/protocol_api/instrument_context.py [hunk 2]
```diff
@@ -75,7 +75,6 @@ def __init__(self,
         self._last_location: Union[Labware, Well, None] = None
         self._last_tip_picked_up_from: Union[Well, None] = None
         self._log = log_parent.getChild(repr(self))
-        self._log.info("attached")
```
Comment (b-cooper) [inline api/src/opentrons/protocol_api/instrument_context.py]: Also noticed this leftover debug log.

**Review by sfoster1 — APPROVED** [review]
Code looks good, if all the tests pass I think we're good!

**Review by shlokamin — APPROVED** [review]

**Comment by b-cooper** [comment]
Tested on bitterwood and OT2CEP20190109A02 with @nusrat813:

- Boot up bot
- simulate  smoke test
- tip probe and labware cal for smoke test 
- run smoke test protocol
- gui deck cal
- cli deck cal
- Jupyter protocol (modules still borked for apiv2)

**Pull request merged**